### ISSUE TITLE
aws cloudformation multiple --parameters flags

### ISSUE THREAD

**Issue by neilhwatson** [aws/aws-cli#3251]
It should be possible to

```
aws cloudformation create-stack --parameters file://p.json --parameters "ParameterKey=AdditionaParam,ParameterValue=some_value"
```

Such that one can provide parameters from both a file and in line. It seems that is not possible now, but sadly the command does not exit with an error stating so. Either add the feature or make an error handler.

**Comment by stealthycoin** [comment]
I will mark this as a feature request, none of the other parameters behave this way so I don't think it should be expected that it would work this way, but I do see it could be quite useful if you wanted defaults loaded from a file and then you could provide specific overrides. This would require a customization on top of the API for this particular case and wouldn't be backwards compatible (the error or the feature itself).

**Comment by rahulakash** [comment]
This definitely would be a great feature to have. Coming from Kubernetes & Helm world, this is something that is expected to be supported by default.

**Comment by whereisaaron** [comment]
Or else just multiple file parameters the same way literal parameters work?
`--parameters file://p.json,file://q.json`

**Comment by shqear-mawdoo3** [comment]
any updates?

**Comment by jiang-gao** [comment]
any updates?

**Comment by tculp** [comment]
Over 2 years out now, any updates?

**Comment by lqueryvg** [comment]
Any news ?

**Comment by lqueryvg** [comment]
I'd like to see multiple `--parameters` flags supported, and whether the parameter values are literal (values specified on the command line) or come from config files, they should be processed from left to right and "compiled" into one final set of parameters which gets passed into the stack.

If a value is specified more than once, the one specified last should win.

### Use case 1:

This gives the user the power to take a layered approach to keep their configuration flexible and DRY.
Two tools which spring to mind are [heira](https://puppet.com/docs/puppet/7.6/hiera_intro.html) and  [helm](https://helm.sh/docs/intro/using_helm)

Quoting from the `helm` usage:
> There are two ways to pass configuration data during install:
> --values (or -f): Specify a YAML file with overrides. This can be specified multiple times and the rightmost file will take precedence

### Use case 2:

The bulk of your configuration may be in a parameters file, but you may have some sensitive values which you are happy to specify on the command line but you don't want to specify in the file in case it accidentally gets committed to git.
In this case it would be better if you could add those to the CLI as separate `--parameters` flags.

**Comment by BenMarshallTR** [comment]
yes please to both multiple parameter files and cli override/additional options

**Comment by tim-finnigan** [comment]
I think there is some overlap here with this request: https://github.com/aws/aws-cli/issues/2429
